plugins/watcher/cmd/insights-event-watcher/main.go

```diff
@@ -0,0 +1,322 @@
+		}
+		if _, err := url.Parse(insightsHost); err != nil {
+			return fmt.Errorf("invalid insights host URL '%s': %w", insightsHost, err)
+		}
+	}
+
+	// Validate organization name if provided
+	if organization != "" {
+		if strings.TrimSpace(organization) == "" {
+			return fmt.Errorf("organization name cannot be empty or whitespace only")
+		}
+		if len(organization) > 100 {
+			return fmt.Errorf("organization name too long (max 100 characters)")
+		}
+	}
+
+	// Validate cluster name if provided
+	if cluster != "" {
+		if strings.TrimSpace(cluster) == "" {
+			return fmt.Errorf("cluster name cannot be empty or whitespace only")
+		}
+		if len(cluster) > 100 {
+			return fmt.Errorf("cluster name too long (max 100 characters)")
+		}
+	}
+
+	// Note: Token validation is now handled via environment variables
+
+	// Validate audit log path if provided
+	if auditLogPath != "" {
+		if strings.TrimSpace(auditLogPath) == "" {
+			return fmt.Errorf("audit log path cannot be empty or whitespace only")
+		}
+		// Check if file exists and is readable
+		if _, err := os.Stat(auditLogPath); err != nil {
+			return fmt.Errorf("audit log path '%s' is not accessible: %w", auditLogPath, err)
+		}
+	}
+
+	// Validate event buffer size
+	if eventBufferSize < 1 {
+		return fmt.Errorf("event buffer size must be at least 1, got %d", eventBufferSize)
+	}
+	if eventBufferSize > 10000 {
+		return fmt.Errorf("event buffer size too large (max 10000), got %d", eventBufferSize)
+	}
+
+	// Validate HTTP timeout
+	if httpTimeoutSeconds < 1 {
+		return fmt.Errorf("HTTP timeout must be at least 1 second, got %d", httpTimeoutSeconds)
+	}
+	if httpTimeoutSeconds > 300 {
+		return fmt.Errorf("HTTP timeout too large (max 300 seconds), got %d", httpTimeoutSeconds)
+	}
+
+	// Validate rate limit
+	if rateLimitPerMinute < 1 {
+		return fmt.Errorf("rate limit must be at least 1 call per minute, got %d", rateLimitPerMinute)
+	}
+	if rateLimitPerMinute > 1000 {
+		return fmt.Errorf("rate limit too high (max 1000 calls per minute), got %d", rateLimitPerMinute)
+	}
+
+	// Validate log source
+	validLogSources := []string{"local", "cloudwatch"}
+	logSourceValid := false
+	for _, valid := range validLogSources {
+		if logSource == valid {
+			logSourceValid = true
+			break
+		}
+	}
+	if !logSourceValid {
+		return fmt.Errorf("invalid log source '%s', must be one of: %s", logSource, strings.Join(validLogSources, ", "))
+	}
+
+	// Validate CloudWatch configuration if log source is cloudwatch
+	if logSource == "cloudwatch" {
+		if cloudwatchLogGroup == "" {
+			return fmt.Errorf("cloudwatch-log-group is required when log-source is cloudwatch")
+		}
+		if cloudwatchRegion == "" {
+			return fmt.Errorf("cloudwatch-region is required when log-source is cloudwatch")
+		}
+		if !strings.HasPrefix(cloudwatchLogGroup, "/aws/eks/") {
+			return fmt.Errorf("cloudwatch-log-group should start with '/aws/eks/', got '%s'", cloudwatchLogGroup)
+		}
+	}
+
+	// Validate CloudWatch batch size
```

Comment:
shouldn't all CloudWatch validations be inside `if logSource == "cloudwatch"` block?

Reply:
done